mlx/primitives.cpp

```diff
@@ -1338,6 +1338,66 @@ std::pair<array, int> LogicalNot::vmap(
   return {logical_not(inputs[0], stream()), axes[0]};
 }
 
+std::vector<array> LogicalAnd::vjp(
+    const std::vector<array>& primals,
+    const array& cotan,
+    const std::vector<int>& argnums) {
+  assert(primals.size() == 2);
+  assert(argnums.size() == 2);
+  return {jvp(primals, {cotan, cotan}, argnums)};
+}
+
+array LogicalAnd::jvp(
+    const std::vector<array>& primals,
+    const std::vector<array>& tangents,
+    const std::vector<int>& argnums) {
+  assert(primals.size() == 2);
+  assert(tangents.size() == 2);
+  assert(argnums.size() == 2);
+  // Gradient of AND is zero w.r.t. both inputs when both are true or false.
+  // When they differ, the gradient is non-zero only w.r.t. the true input.
+  return logical_and(primals[0], tangents[1], stream()) +
+      logical_and(tangents[0], primals[1], stream());
+}
+
+std::pair<array, int> LogicalAnd::vmap(
+    const std::vector<array>& inputs,
+    const std::vector<int>& axes) {
+  assert(inputs.size() == 2);
+  assert(axes.size() == 2);
+  return {logical_and(inputs[0], inputs[1], stream()), axes[0]};
```

Comment:
This is a bug, can you add a test for it? Binary operations need to make sure the axes are in the same place for the vmap. See `Add` or `Multiply` for an example of how this should go.